librecad/src/lib/math/rs_math.h

```diff
@@ -36,117 +36,122 @@ class QString;
-      *@return a RS_VectorSolutions contains real roots (x,y)
-      */
-    static RS_VectorSolutions simultaneousQuadraticSolverFull(const std::vector<std::vector<double> >& m);
-    static RS_VectorSolutions simultaneousQuadraticSolverMixed(const std::vector<std::vector<double> >& m);
-
-	/** \brief verify simultaneousQuadraticVerify a solution for simultaneousQuadratic
-	  *@param m the coefficient matrix
-	  *@param v a candidate to verify
-      *@return true, for a valid solution
-      **/
-	static bool simultaneousQuadraticVerify(const std::vector<std::vector<double> >& m, RS_Vector& v);
-    /** wrapper for elliptic integral **/
-    /**
-     * wrapper of elliptic integral of the second type, Legendre form
-     *@k the elliptic modulus or eccentricity
-     *@phi elliptic angle, must be within range of [0, M_PI]
-     *
-	 *@\author: Dongxu Li
-     */
-    static double ellipticIntegral_2(const double& k, const double& phi);
-
-    static QString doubleToString(double value, double prec);
-    static QString doubleToString(double value, int prec);
-
-    static void test();
+class RS_Math
+{
+    private:
+        RS_Math() = delete;
+
+    public:
+        static int round(double v);
+        static double pow(double x, double y);
+        static RS_Vector pow(RS_Vector x, double y);
+        static bool equal(const double d1, const double d2);
+
+        static double rad2deg(double a);
+        static double deg2rad(double a);
+        static double rad2gra(double a);
+        static double gra2rad(double a);
+        static unsigned findGCD(unsigned a, unsigned b);
+        static bool isAngleBetween(double a,
+                                   double a1, double a2,
+                                   bool reversed = false);
+        //! \brief correct angle to be within [0, +PI*2.0)
+        static double correctAngle(double a);
+        //! \brief correct angle to be within [-PI, +PI)
+        static double correctAngle2(double a);
+        //! \brief correct angle to be unsigned [0, +PI)
+        static double correctAngleU(double a);
+
+        //! \brief angular difference
+        static double getAngleDifference(double a1, double a2, bool reversed = false);
+        /**
+         * @brief getAngleDifferenceU abs of minimum angular difference, unsigned version of angular difference
+         * @param a1,a2 angles
+         * @return the minimum of angular difference a1-a2 and a2-a1
+         */
+        static double getAngleDifferenceU(double a1, double a2);
+        static double makeAngleReadable(double angle, bool readable=true,
+                                        bool* corrected=nullptr);
+        static bool isAngleReadable(double angle);
+        static bool isSameDirection(double dir1, double dir2, double tol);
+
+        //! \{ \brief evaluate a math string
+        static double eval(QString expr, double def=0.0);
+        static double eval(QString expr, bool* ok);
+        //! \}
```

Comment:
The changes in the above lines 79 and 80 are the only real changes I made in this code.